api/app/tests/apps/api/test_api_v1_ml_predict_endpoint.py

```diff
@@ -0,0 +1,138 @@
+from unittest.mock import patch
+
+from django.http import Http404
+from rest_framework.exceptions import APIException
+from rest_framework.reverse import reverse
+
+from signals.apps.signals.models import Category
+from signals.apps.signals.models.category_translation import CategoryTranslation
+from tests.apps.signals.factories import CategoryFactory
+from tests.test import SignalsBaseApiTestCase
+
+
+class TestMlPredictCategory(SignalsBaseApiTestCase):
+    test_host = 'http://testserver'
+    endpoint = '/signals/v1/public/category/predict'
+
+    def setUp(self):
+        self.test_parentcategory_overig = Category.objects.get(
+            slug='overig',
+            parent__isnull=True,
+        )
+        self.test_subcategory_overig = Category.objects.get(
+            parent=self.test_parentcategory_overig,
+            slug='overig',
+            parent__isnull=False,
+        )
```

Comment:
Create test categories, do not re-use existing ones. (These may or may not exist in the future.)

Reply:
In this case only the "overig" is used. All others are created. If for some reason "overig" does not exists anymore a code change is needed because this is the default category that will be returned of the ML tool will not answer.